view/src/components/filters.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+
+import axios from "axios";
+
+class Filters extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            locationId: "",
+            distance: "0",
+            days: "0",
+            hours: "0",
+            minutes: "0",
+            results: [],
+        }
+    }
+
+    handleChange = (event) => {
+        this.setState({
+            [event.target.name]: event.target.value,
+        });
+    };
+
+    filterByDistance = () => {
+        let url = "";
+        switch (this.props.database) {
+            case "foodbanks":
+                url = "/queryFoodBanksByDistance";
+                break;
+            case "farms":
+                url = "/queryFarmsByDistance";
+                break;
+            case "surplus":
+                url = "/querySurplusByDistance";
+                break;
+        }
+        url += `?locationId=${this.state.locationId}&distance=${this.state.distance}`;
+        axios.get(url)
+        .then(res => {
+            this.setState({results: res.data});
+        });
+    };
+
+    filterByTravelTime = () => {
+        let url = "";
+        switch (this.props.database) {
+          case "foodbanks":
+            url = "/queryFoodBanksByTravelTime";
+            break;
+          case "farms":
+            url = "/queryFarmsByTravelTime";
+            break;
+          case "surplus":
+            url = "/querySurplusByTravelTime";
+            break;
+        }
+        url += `?locationId=${this.state.locationId}&days=${this.state.days}&hours=${this.state.hours}&minutes=${this.state.minutes}`;
+        axios.get(url)
+        .then(res => {
+            this.setState({results: res.data});
+        });
+    };
+
+    render() {
+        return (
+            <div>
+                <TextField id="outlined-basic" label="Outlined" variant="outlined" label="Location ID" name="locationId" value={this.state.locationId} onChange={this.handleChange} />
```

Comment:
It's not valid to have two elements with the same `id`.  It sounds like maybe you're treating this more like a class name?  But you don't seem to be using it.  Probably just remove.